playbooks/scenario44_rds_instance.yaml

```diff
@@ -48,50 +23,93 @@
+        datastore: "{{ db_type }}"
+        instance_mode: "single"
       register: rds_flavors
 
-    # Print the first flavor in the query
-    - name: debug
-      ansible.builtin.debug:
-        msg: "{{ rds_flavors.rds_flavors[0].name }}"
-     
-    - debug:
-        msg: "Using prefix {{ prefix }}"
-   
+    # Find flavor with smallest RAM and smallest VCPUs
+    - name: Set fact for smallest flavor based on RAM and VCPUs
+      set_fact:
+        smallest_flavor: "{{ rds_flavors.rds_flavors | sort(attribute='ram') | first }}"
+
+    # Run python script to get Region, Storage type and Latest version of DB engine
+    - name: Run Python script to find the latest version
+      script: "{{ playbook_dir }}/files/rds_preconditions.py '{{ db_type }}' '{{ versions.rds_datastores | to_nice_json }}' '{{ smallest_flavor.name }}'"
+      args:
+        executable: python3
+      register: script_output
+
+    - name: Parse JSON output from script
+      set_fact:
+        script_vars: "{{ script_output.stdout | from_json }}"
 
+    - name: Set region, storage_type, and latest_version
+      set_fact:
+        region: "{{ script_vars.region }}"
+        storage_type: "{{ script_vars.storage_type }}"
+        latest_version: "{{ script_vars.latest_version }}"
+
+    - name: Display region and storage_type
+      debug:
+        msg:
+          - "Prefix: {{ prefix }}"        
+          - "Region: {{ region }}"
+          - "Storage Type: {{ storage_type }}"
+          - "DB Type: {{ db_type }}"          
+          - "Latest vesion: {{ latest_version }}"
+          - "Smallest flavor: {{ smallest_flavor.name }}"
+
+   
     - block:
-      # Create VPC and SUBNET
-      - name: Create VPC (Router + Net + Subnet)
+      # Check if the VPC (Router + Network + Subnet) exists
+      - name: Check if Network exists
+        openstack.cloud.networks_info:
+          name: "{{ test_network_name }}"
+        register: network_check
+
+      # Create VPC and Subnet only if the network doesn't exist
+      - name: Create VPC (Router + Net + Subnet) if it doesn't exist
         include_role:
           name: opentelekomcloud.vpc
         vars:
           router_name: "{{ test_router_name }}"
           network_name: "{{ test_network_name }}"
           subnet_name: "{{ test_subnet_name }}"
           state: present
+        when: network_check.networks | length == 0
+
+      # Check if the Security Group exists
+      - name: Check if Security Group exists
+        openstack.cloud.security_group_info:
+          name: "{{ test_security_group_name }}"
+        register: sg_check
 
-      # Create Security Group
-      - name: Create SecurityGroup
+      # Create Security Group only if it doesn't exist
+      - name: Create Security Group if it doesn't exist
         openstack.cloud.security_group:
           name: "{{ test_security_group_name }}"
           description: RDS test SG created by APImon
+        when: sg_check.security_groups | length == 0
```

Comment:
is this necessary?
If the security group already exists then ansible module should automatically do nothing, and should not require any further condition

Reply:
fixed